cmd/zpool/zpool_main.c

```diff
@@ -2521,6 +2522,36 @@ do_import(nvlist_t *config, const char *newname, const char *mntopts,
 	return (ret);
 }
 
+static int
+do_write_config(nvlist_t *config, const char *path)
+{
+	size_t nvsize;
+	char *packed;
+	int fd;
+
+	fd = open(path, O_RDWR | O_CREAT | O_TRUNC, 0644);
+	if (fd < 0) {
+		(void) fprintf(stderr, gettext("can't open '%s': %s\n"),
+		    path, strerror(errno));
+		return (errno);
+	}
+	packed = fnvlist_pack(config, &nvsize);
+
+	char *bufptr = packed;
+	int residual = nvsize;
+
+	while (residual > 0) {
+		ssize_t bytes = write(fd, bufptr, residual);
+		ASSERT(bytes >= 0);
```

Comment:
This could fail with `ENOSPC` or potentially other errnos.  Can you add some error handling to cleanly handle something unexpected and log the error.

Reply:
Added error handling for write failures